constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/helper/widget/Grid.java

```diff
@@ -575,4 +667,20 @@ private float[] getLinspace(float min, float max, int positions) {
         }
         return d;
     }
+
+    public void setStrSpans(String strSpans) {
+        mStrSpans = strSpans;
+    }
+
+    public String getStrSpans() {
+        return mStrSpans;
+    }
+
+    public void setStrSkips(String strSkips) {
+        mStrSkips = strSkips;
+    }
+
```

Comment:
As above 
The UI will not change if I set a different span
Create a private methods to update the UI when setStrSpans;
setStrSpans is a bad name it should be like setSpans(String spanStr)

No Java doc